### ISSUE TITLE
aws_stepfunctions_tasks.EcsRunTask: Resource not being created

### ISSUE THREAD

**Issue by automartin5000** [aws/aws-cdk#18337]
### What is the problem?

When I create a `EcsRunTask` resource, it strangely only seems to create the security group (as well as the DependsOn attributes and security group ingress/egress entries) and not the actual Step Functions job. There's no errors. The stack synths and deploys correctly. It just has no Step Functions resource, just the the security group.

Removing the dependency and the security group connections, leaves just the security group resource. Very strange (see code).

### Reproduction Steps

```python
self.task_runner_params = {
    "integration_pattern": sfn.IntegrationPattern.RUN_JOB,
    "cluster": self.ecs_cluster,
    "task_definition": self.task,
    "launch_target": task_runner.EcsFargateLaunchTarget(platform_version=ecs.FargatePlatformVersion.LATEST)
}
self.run_task = task_runner.EcsRunTask(self, "RunTask",
    **self.task_runner_params,
    container_overrides=[task_runner.ContainerOverride(
        container_definition=self.container,
        command=["python3"]
    )],
)

self.services_to_connect_to_network_dependencies += [self.run_task.connections]
self.scope.compute_stack.app_service.node.add_dependency(self.run_task)
```

### What did you expect to happen?

Step Functions job resource is created and run.

### What actually happened?

Security group is created but not the EcsRunTask resource. No error.

### CDK CLI Version

2.5

### Node.js Version

16.13.1

### OS

Mac OS 12.1

### Language

Python

### Language Version

3.10.1

### Other information

Removing the `add_dependency` and security group connections removes those as expected, just leaving the security group as the sole resource created.

**Comment by kaizen3031593** [comment]
A StepFunctions task is meant to be added as part of a StepFunctions StateMachine definition. The task alone does not map to any particular resource. Are you also creating a StepFunctions StateMachine that uses the task as part of the definition? 

Here's an example of how you can use `EcsRunTask` as part of a state machine:

https://github.com/aws/aws-cdk/blob/a8094c7d9970557077f560ccd24882216094ee3c/packages/%40aws-cdk/aws-stepfunctions-tasks/test/ecs/run-tasks.test.ts#L130

Here's an integ test for `EcsRunTask`.

https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-stepfunctions-tasks/test/ecs/integ.fargate-run-task.ts

tl:dr; `EcsRunTask` is not a resource itself. It gets rendered as part of a state machine definition.

Let me know if this helps!
